crates/iota-genesis-builder/src/stardust/migration/migration.rs

```diff
@@ -135,7 +142,7 @@ impl Migration {
             .collect::<Vec<_>>();
         info!("Verifying ledger state...");
         self.verify_ledger_state(&outputs)?;
-
+        self.address_swap_map.verify_all_addresses_swapped();
```

Comment:
inside this function, instead of panic could you return an `anyhow::Result` and throw `anyhow!("address to swap not found:  {origin_address}")`

Reply:
Done